File: Web/Plugins/Lemoine.Plugin.MachiningProcessMilestones/WebExtension.fs
(*
 * Copyright (C) 2009-2023 Lemoine Automation Technologies
 *
 * SPDX-License-Identifier: Apache-2.0
 *)

namespace Lemoine.Plugin.MachiningProcessMilestones

#if !NSERVICEKIT
(* Note: this does not work currently with the web service, see story #175262470 *)

type WebExtension() =

  member val UniqueInstance = true
  member this.GetAssembly () =
    this.GetType().Assembly

  interface Lemoine.Extensions.IExtension with
    member this.UniqueInstance = this.UniqueInstance

  interface Lemoine.Extensions.Web.IWebExtension with
    member this.GetAssembly () = this.GetAssembly ()

#endif // !NSERVICEKIT

File: Plugins/Lemoine.Plugin.MaintenanceAction/WebExtension.cs
// Copyright (C) 2009-2023 Lemoine Automation Technologies
//
// SPDX-License-Identifier: Apache-2.0

using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Text;

namespace Lemoine.Plugin.MaintenanceAction
{
  public class WebExtension
    : Lemoine.Extensions.Web.IWebExtension
  {
    public bool UniqueInstance
    {
      get
      {
        return true;
      }
    }

    public Assembly GetAssembly ()
    {
      return this.GetType ().Assembly;
    }
  }
}
